custom/covid/management/commands/data_cleaning_patient_closed.py

```diff
@@ -0,0 +1,46 @@
+from xml.etree import cElementTree as ElementTree
+
+from django.core.management.base import BaseCommand
+
+from casexml.apps.case.mock import CaseBlock
+from dimagi.utils.chunked import chunked
+
+from corehq.apps.es import CaseSearchES
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.apps.users.util import SYSTEM_USER_ID, username_to_user_id
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+
+BATCH_SIZE = 100
+DEVICE_ID = __name__ + ".update_closed"
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('--username', type=str, default=None)
+
+    def handle(self, domain, **options):
+        if options["username"]:
+            user_id = username_to_user_id(options["username"])
+            if not user_id:
+                raise Exception("The username you entered is invalid")
+        else:
+            user_id = SYSTEM_USER_ID
+
+        case_ids = CaseSearchES().domain(domain).xpath_query(domain, "closed != ''").values_list("_id", flat=True)
+        accessor = CaseAccessors(domain)
+        case_blocks = []
+        for case in accessor.iter_cases(case_ids):
+            case_blocks.append(ElementTree.tostring(CaseBlock.deprecated_init(
+                create=False,
+                case_id=case.case_id,
+                update={"closed": ""},
```

Comment:
should this be close=False instead? Assuming that "update" is passed for case properties and close needs to be passed separately.

or we need to determine if case was closed before and then keep it closed else open? If needed, could use `corehq.form_processor.models.CommCareCaseSQL.get_closing_transactions`

Reply:
my understanding of this is that there is a case property called `closed` which has nothing to do with whether or not the case state is set to `closed`

Reply:
oh cool. I think i am confused by https://dimagi-dev.atlassian.net/browse/SUPPORT-9422?focusedCommentId=143952 and the comment made after that. If we need to clean just the case property, this all looks good.

Reply:
Yeah, as @snopoke said, these cases had a `closed` case property, which was problematic. But I added d7cb129b084 to update this to set the `closed` case property to the value of `case.closed` (which correctlys return a boolean, even for the problematic cases) instead of a blank string.